### ISSUE TITLE
ObjectBoxLiveData doesn't emit properly while updating multiple objects at same time

### ISSUE THREAD

**Issue by prvnmit** [objectbox/objectbox-java#540]
## Issue Basics
- ObjectBox version (are using the latest version?): 2.1.0
- Reproducibility: always

### Description
While updating multiple objects together, LiveData just emits for the first update, and not for the rest. 

### Code
        // Observing themes
        viewModel.getThemes().observe(this, Observer { themes ->
            if(themes != null) adapter.submitList(ArrayList(themes))
        })
        // Updating 2 theme objects
        prevTheme?.isCurrentTheme = false
        currentTheme.isCurrentTheme = true
        boxStore.runInTx {
            prevTheme?.let { themesBox.put(it) }
            currentTheme.let { themesBox.put(it) }
        }

## Misc
The same occurs when the 2 updates are done without a transaction/in a single 'put' call
I would appreciate if LiveData could emit once per transaction for the above scenario, since the emission is used for updating a recyclerview and it makes sense to do it once the entire transaction is complete.

**Comment by greenrobot-team** [comment]
In both cases there should only be a single notification.

Anyhow, I just tried to reproduce this using [our `objectbox-arch-example`](https://github.com/objectbox/objectbox-examples/tree/master/objectbox-arch-example) by adding a log statement to the `onChanged()` callback of the observer:
```java
model.getNoteLiveData(notesBox).observe(this, new Observer<List<Note>>() {
    @Override
    public void onChanged(@Nullable List<Note> notes) {
        Log.d(TAG, "onChanged() called with: notes = [" + notes + "]");
    }
});
```

Then I just added two notes, once with
`notesBox.put(note1, note2);`
and then with
```java
boxStore.runInTx(new Runnable() {
    @Override
    public void run() {
        notesBox.put(note1);
        notesBox.put(note2);
    }
});
```
In both cases `onChanged()` is only called once.

As you are using `adapter.submitList()` maybe you are using `ObjectBoxDataSource` and not `ObjectBoxLiveData`, so tried the above using the data source example and paging library.

However, again, `onChanged()` is only called once.

So, can you please share a small example project that reproduces this issue?
-ut

**Comment by greenrobot-team** [comment]
If you get the correct list in `onChanged()` and it is modified before/while `submitList()` is called I do not follow how ObjectBox is at fault?

```java
liveData.observe(this, new Observer<List<Theme>>() {
    @Override
    public void onChanged(@Nullable List<Theme> items) {
        // list is correct here? if yes, ObjectBox works as expected
       adapter.submitList(list);
    }
});
```

Check how you keep references to the old and new list as well as if you hold onto any list items.
-ut

**Comment by greenrobot-team** [comment]
No response, closing. Please re-open with more details or submit a new issue.
-ut
